enginetest/insert_queries.go

```diff
@@ -1167,38 +1167,110 @@ var InsertIgnoreScripts = []ScriptTest{
 		},
 	},
 	{
-		Name: "Test that INSERT IGNORE INTO works with unique keys",
+		Name: "Test that INSERT IGNORE properly addresses data conversion",
```

Comment:
You need a lot more tests of this behavior with lots of different types and values. I would start with unit tests, then move on to engine tests

Reply:
Zach, thank you for this comment. I appreciate the attention to extra robustness in the fix, and that you added tests that make it unlikely to break again.